### ISSUE TITLE
quaternion2euler doesn't work for certain sets of angles

### ISSUE THREAD

**Issue by Fifourche** [napari/napari#2525]
## 🐛 Bug

In 3D viewer mode, if ``viewer.camera.angles`` is set to some particular set of Euler angles, a RuntimeWarning is raised and the ``viewer.camera.angles`` sends back a tuple containing a ``nan``.

We can trace it back to the ``quaternion2euler`` function that doesn't work as intended. For certain angles, the argument that goes into the ``arcsin`` function is greater than 1, raising an error and returning a tuple containing a ``nan``.

## To Reproduce

Steps to reproduce the behavior via the **viewer** (easy to show how problematic that can be) :

1. Set ``viewer.camera.angles`` to a particular angle -> raises a warning
2. Ask back ``viewer.camera.angles`` -> shows the ``nan``

```Python
import napari
viewer = napari.Viewer()

viewer.dims.ndim = 3
viewer.dims.ndisplay = 3
viewer.axes.visible = True

# Particular set of angles
angles_0 = (180,-90,0)

# Set view
viewer.camera.angles = angles_0
# Get view and print
print(viewer.camera.angles)
```

Steps to reproduce the behavior directly via the function ``quaternion2euler`` :

1. Convert the set of angles to a vispy ``Quaternion`` (as done in the camera)
2. Convert back this object to Euler angles via the ``quaternion2euler`` function

```Python
from napari._vispy.quaternion import quaternion2euler
from vispy.util.quaternion import Quaternion as Q

# Particular set of angles
angles_0 = (180,-90,0)

quaternion = Q.create_from_euler_angles(*angles_0, degrees = True)

# Supposedly same angles...
angles = quaternion2euler(quaternion, degrees = True)

print(str(angles_0) + ' became ' + str(angles))
```

The warning that is raised is :

``..\napari\_vispy\quaternion.py:128: RuntimeWarning: invalid value encountered in arcsin
  np.arcsin(2 * (q.w * q.y - q.z * q.x)),``

## Expected behavior

Expected to get back the exact tuple of angles we just set and without warning.

## Environment

napari: 0.4.7
Platform: Windows-10-10.0.19041-SP0
Python: 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 15:50:08) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.3
NumPy: 1.20.1
SciPy: 1.6.1
Dask: 2021.02.0
VisPy: 0.6.6

OpenGL:
- GL version: 4.6.0 NVIDIA 457.09
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 1920x1080, scale 1.0

Plugins:
- animation: 0.1.dev86+g1fa7b65
- console: 0.0.3
- svg: 0.1.4

## Additional context

The problem is half solved if we cap the modulus of the ``arcsin`` argument by 1, but then there is still a problem in the conversion leading to "get" angles different from the ones we set. I'll probably open another issue for that.

**Comment by sofroniewn** [comment]
Wow, nice catch!! Thanks for filing this report and the PR, I'll take a look at that now too. We'll definitely want to fix this

**Comment by sofroniewn** [comment]
This was closed by #2530
